lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -41,7 +49,12 @@ def on_send(node)
         def yoda_condition?(node)
           return false unless node.comparison_method?
 
-          node.receiver.literal? && !node.arguments.first.literal?
+          first, operator, last = node.children
+          if check_equality_only?
+            return false if non_equality_operator?(operator)
```

Comment:
@Drenmi that was my initial idea, but I didn't like how it reads. So I changed it. The code I have now reads a little bit better, but I feel like it's a matter of preference.